src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,69 @@
+import { CommentType } from '../../types/Comment';
+import { PostType } from '../../types/Post';
+import { CommentsList } from '../CommentsList/CommentsList';
+import { useComments } from '../Contexts/CommentsContext';
+import { NewCommentForm } from '../NewCommentForm/NewCommentForm';
+
+type Props = {
+  onAddComment: (status: boolean) => void,
+  onDeleteComment: (id: number) => void,
+  comments: CommentType[],
+  isNewComment: boolean,
+  hasError: boolean,
+  selectedPost: PostType,
+};
+
+export const PostDetails: React.FC<Props> = ({
+  onAddComment,
+  onDeleteComment,
+  comments,
+  isNewComment,
+  hasError,
+  selectedPost,
+}) => {
+  const { setComments } = useComments();
+
+  const toggleAddCommentForm = () => {
+    onAddComment(true);
+  };
+
+  const handleDeleteComment = (commentId: number) => {
+    setComments(prev => prev.filter(({ id }) => id !== commentId));
+    try {
+      onDeleteComment(commentId);
+    } catch {
```

Comment:
```suggestion
  const handleDeleteComment = async (commentId: number) => {
    setComments(prev => prev.filter(({ id }) => id !== commentId));
    try {
      await onDeleteComment(commentId);
    } catch {
```
`onDeleteComment` is an async operation so you should wait for the result.

For this task it's okay, but in real life, if something goes wrong you should notify the user and get back previous data, for example, if a comment wasn't added successfully, you should remove it from the UI. The same if the deletion fails you should get back the comment and show it, and notify a user that the error occurred